pkg/recipes/driver/bicep.go

```diff
@@ -162,6 +163,12 @@ func (d *bicepDriver) Execute(ctx context.Context, opts ExecuteOptions) (*recipe
 
 	resp, err := poller.PollUntilDone(ctx, &runtime.PollUntilDoneOptions{Frequency: pollFrequency})
 	if err != nil {
+		errorDetails := recipes.GetErrorDetails(err)
+		for _, error := range errorDetails.Details {
+			if error.Code == "AuthorizationFailed" && providerConfig.Az == nil {
+				return nil, v1.NewClientErrInvalidRequest("Azure deployment failed due to authorization error, please ensure you have configured an Azure provider with your Radius environment: https://docs.radapp.io/guides/operations/providers/azure-provider/")
+			}
+		}
 		return nil, recipes.NewRecipeError(recipes.RecipeDeploymentFailed, fmt.Sprintf("failed to deploy recipe %s of type %s", opts.BaseOptions.Recipe.Name, opts.BaseOptions.Definition.ResourceType), recipes_util.ExecutionError, recipes.GetErrorDetails(err))
```

Comment:
Would this error message be displayed only when a bicep recipe deploying an azure resource fails due to missing provider?

Reply:
Do we want to say more about the authorization error, I see a lot because our IT department likes to play MITM, maybe distinguish something like a certificate error to other flavours of error?

Reply:
@vishwahiremat and I discussed and decided to move this fix to the deployment engine. https://github.com/radius-project/deployment-engine/pull/395

It's better if the DE reports this error with it's deep knowledge of the scenario rather than trying to guess the problem from the CLI.

Reply:
@superbeeny Here is the PR for this fix in Deployment engine: https://github.com/radius-project/deployment-engine/pull/395.